include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -197,6 +197,102 @@ class guard_view
     diff_type m_count; // size of the data
 };
 
+//It is kind of pseudo-view for reverse_view support.
+template <typename _R, typename _Size>
+struct reverse_view_simple
+{
+    _R __r;
+    _Size __n;
+
+    template <typename Idx>
+    auto operator[](Idx __i) const -> decltype(__r[__i])
+    {
+        return __r[__n - __i - 1];
```

Comment:
Will it be correct if we use `__r.size()` instead of `__n`. In that case we don't need an extra parameter for struct.

Reply:
yeap...